src/components/shared/CategoryFilter.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { BsChevronCompactLeft, BsChevronCompactRight } from 'react-icons/bs';
+import { v4 as uuidv4 } from 'uuid';
+import { ProductAPIDataProps } from '../../utils/dummyCategories';
+
+type categoriesProps = {
+  categories: string[];
+  apiData: ProductAPIDataProps;
+};
+
+function CategoryFilter({ categories, apiData }: categoriesProps) {
+  const [activeItem, setActiveItem] = useState(0);
+
+  let result = apiData.filter(
+    (curProduct) => curProduct.category === categories[0],
+  );
+
+  const [data, setData] = useState(result);
+
+  const filterResult = (category: string) => {
+    result = apiData.filter((curProduct) => curProduct.category === category);
+    setData(result);
+  };
+
+  // Category Onclick event handler for mobile devices
+  const handlePrev = () => {
+    const isFirstCategory = activeItem === 0;
+    const newIndex = isFirstCategory ? categories.length - 1 : activeItem - 1;
+    setActiveItem(newIndex);
+
+    result = apiData.filter(
+      (curProduct) => curProduct.category === categories[newIndex],
+    );
+    setData(result);
+  };
+
+  const handleNext = () => {
+    const isLastSlide = activeItem === categories.length - 1;
+    const newIndex = isLastSlide ? 0 : activeItem + 1;
+    setActiveItem(newIndex);
+
+    result = apiData.filter(
+      (curProduct) => curProduct.category === categories[newIndex],
+    );
+    setData(result);
+  };
+
+  return (
+    <div className="flex flex-col justify-center items-center">
+      <div className="relative hidden sm:flex flex-col sm:flex-row justify-between bg-white shadow p-1 border-2 rounded-lg border-gray-500">
```

Comment:
Also, this border is taking away rather than adding to the design or looks of this component, you should either reduce it's size, color etc....